### PR TITLE
[OV]: Fixed inference after 4 bit weight compression

### PR DESCRIPTION
**Pull request by AlexKoff88**
Fixed inferencing the baseline model instead of the quantized version (delayed compilation).

### CODE REVIEW

**Review by eaidova — APPROVED** [review]

File: optimum/intel/openvino/modeling_decoder.py [hunk 1]
```diff
@@ -592,7 +592,9 @@ def _from_pretrained(
         else:
             init_cls = cls
 
-        causal_model = init_cls(model=model, config=config, model_save_dir=model_cache_path.parent, **kwargs)
+        causal_model = init_cls(
+            model=model, config=config, model_save_dir=model_cache_path.parent, compile=not load_in_4bit, **kwargs
+        )
```
Comment (echarlaix) [inline optimum/intel/openvino/modeling_decoder.py:597]:
```suggestion
        enable_compilation = kwargs.pop("compile", True) and not load_in_4bit
        causal_model = init_cls(
            model=model, config=config, model_save_dir=model_cache_path.parent, compile=enable_compilation, **kwargs
        )
```

File: optimum/intel/openvino/modeling_decoder.py [hunk 2]
```diff
@@ -609,6 +611,7 @@ def _from_pretrained(
                 )
 
             _weight_only_quantization(causal_model, quantization_config)
+            causal_model.request = None
```
Comment (echarlaix) [inline optimum/intel/openvino/modeling_decoder.py:614]: why is this needed ?

**Review by echarlaix — APPROVED** [review]
thanks a lot for the fix!

**Comment by echarlaix** [comment]
thanks for iterating @AlexKoff88, a last modification is needed here to support cases where self.request is None https://github.com/huggingface/optimum-intel/blob/8f7d016fbeb0cd185e80b9852207c2b530d3d394/optimum/intel/openvino/modeling_decoder.py#L422
